src/site/content/en/blog/granular-chunking-nextjs/index.md

```diff
@@ -0,0 +1,273 @@
+
+1.  **Page load times are improved**. Emitting multiple shared chunks, instead of a single one,
+    minimizes the amount of unneeded (or duplicate) code for any entrypoint.
+2.  **Improved caching during navigations**. Splitting large libraries and framework dependencies
+    into separate chunks reduces the possibility of cache invalidation since both are unlikely to
+    change until an upgrade is made.
+
+## More HTTP requests
+
+`SplitChunksPlugin` defined the basis for granular chunking, and applying this approach to a
+framework like Next.js was not an entirely new concept. Many frameworks, however, still continued to
+use a single heuristic and "commons" bundle strategy for a few reasons. This includes the concern that
+many more HTTP requests can negatively affect site performance.
+
+Browsers can only open a limited number of TCP connections to a single origin (6 for Chrome), so
+minimizing the number of chunks outputted by a bundler can ensure that the total number of requests
+stays under this threshold. However, this only holds true for HTTP/1.1. Multiplexing in HTTP/2
+allows for multiple requests to be streamed in parallel using a single connection over a single
+origin. In other words, we generally do not need to worry about limiting the number of chunks
+emitted by our bundler.
+
+[All major browsers](https://caniuse.com/#feat=http2) support HTTP/2, and we wanted to see for
+ourselves if increasing the number of requests by splitting Next.js's single "commons" bundle into
+multiple shared chunks would affect loading performance in any way. We began by measuring the
+performance of a single site while modifying the maximum number of parallel requests using the
+[`maxInitialRequests`](https://webpack.js.org/plugins/split-chunks-plugin/#splitchunksmaxinitialrequests) property.
+
+![Page load performance with increased number of requests](./performance-num-requests-1.png)
+
+In an average of three runs of multiple trials on a single web page, the load, start-render and
+first contentful paint times all remained about the same when varying the max initial request count
+(from 5 to 15). Interestingly enough, we noticed a slight performance overhead only after splitting
+aggressively to hundreds of requests.
+
+![Page load performance with hundreds of requests](./performance-num-requests-2.png)
+
+This showed that staying under a reliable threshold (20~25 requests) struck the right balance
+between loading performance and caching efficiency. After some baseline testing, 25 was selected as
+the `maxInitialRequest` count. 
+
+Modifying the maximum number of requests that happen in parallel resulted in more than a single
+shared bundle, and separating them appropriately for each entry point significantly reduced the
+amount of unneeded code for the same page.
+
+![JavaScript payload reductions with increased chunking](./js-payload-num-requests.png)
+
+## Size reductions with granular chunks
```

Comment:
```suggestion
![JavaScript payload reductions with increased chunking](./js-payload-num-requests.png)

In other words, these results suggest that setting `maxInitialRequests` to `25` on the test page
was optimal because it reduced the JavaScript payload size without slowing down page load
performance.

## Size reductions with granular chunks
```

@housseindjirdeh you can rephrase this however you like but I think you should add a paragraph that summarizes the findings.

However, here's a bigger performance narrative thing that I don't understand: we always talk about how JS payload is a big factor in page load performance. I don't understand how the page load stayed the same while the total payload reduced. It seems like they should both go down. I think you should add a paragraph explaining what's going on there.

Reply:
Yeah I was under the impression that this was just to check if performance didn't degrade by increasing the number of requests (and not doing any optimizations yet), but I agree it's a little confusing with the JavaScript reductions.

@atcastle: I took a stab explaining what's happening here. Could you take a quick look and verify?